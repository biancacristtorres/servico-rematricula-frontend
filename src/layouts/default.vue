<template>
  <v-app class="grey lighten-2 app-rematricula">
    <cabecalho :titulo="'Rematrícula'" v-if="!estaNaTelaDeLogin"></cabecalho>

    <v-content>
      <router-view>
      </router-view>
    </v-content>

    <v-btn id="impersonate" v-if="impersonate"
        color="primary"
        fixed
        bottom
        left>
        {{ textoDoBotaoDeImpersonate() }}
    </v-btn>
  </v-app>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { namespace } from 'vuex-class';
import { StoreNamespaces } from '@/store';
import { InformacoesAluno } from '@/model/InformacoesAluno';
import UsuarioIdentity from 'componente-frontend-core/model/UsuarioIdentity';
import Cabecalho from '@/components/Cabecalho.vue';
const autorizacao = namespace(StoreNamespaces.AUTORIZACAO);
const aluno = namespace(StoreNamespaces.ALUNO);

@Component({ components: { Cabecalho }})
export default class LayoutPage extends Vue {

  @autorizacao.State
  public impersonate!: boolean;

  @aluno.State
  public informacoesAluno!: InformacoesAluno;

  private get estaNaTelaDeLogin(): boolean {
    const fullPath = this.$route.fullPath.toLowerCase();
    return fullPath.includes('/login');
  }

  public textoDoBotaoDeImpersonate(): string {
    return `Você está logado como ${ this.informacoesAluno ? this.informacoesAluno.nomeSocial : '' }`;
  }
}
</script>