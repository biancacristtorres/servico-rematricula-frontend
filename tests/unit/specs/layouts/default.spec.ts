import { DefaultFactory } from '../../factory/layouts/DefaultFactory';

describe('layout > Página Default', () => {

  const factory = new DefaultFactory();

  it('Snapshot', async (done) => {
    await factory.montarComponente();

    expect(factory.componente).toMatchSnapshot();
    done();
  });

  it('Exibe um botão com o usuário impersonate', async (done) => {
    factory.dadoQueEImpersonate();
    factory.dadoQueAsInformacoesDoAlunoTenhaNomeSocial();

    await factory.montarComponente();

    const impersonate = factory.componente.find('#impersonate');
    expect(impersonate.exists()).toBeTruthy();
    expect(factory.vueInstance.textoDoBotaoDeImpersonate()).toEqual('Você está logado como Maria');

    done();
  });

  it('Não exibe botão de impersonate', async (done) => {
    factory.dadoQueNaoEImpersonate();

    await factory.montarComponente();

    const impersonate = factory.componente.find('#impersonate');
    expect(impersonate.exists()).toBeFalsy();

    done();
  });
});
