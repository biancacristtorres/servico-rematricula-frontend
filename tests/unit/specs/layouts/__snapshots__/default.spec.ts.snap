// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout > Página Default Snapshot 1`] = `
<vapp-stub id="app" class="grey lighten-2 app-rematricula">
  <cabecalho-stub titulo="Rematrícula"></cabecalho-stub>
  <vcontent-stub tag="main">
    <router-view></router-view>
  </vcontent-stub>
  <!---->
</vapp-stub>
`;
